src/coreclr/src/vm/prestub.cpp

```diff
@@ -1867,14 +1897,50 @@ extern "C" PCODE STDCALL PreStubWorker(TransitionBlock * pTransitionBlock, Metho
     UNINSTALL_MANAGED_EXCEPTION_DISPATCHER;
 
     {
-        HardwareExceptionHolder
+        HardwareExceptionHolder;
 
         // Give debugger opportunity to stop here
         ThePreStubPatch();
     }
 
     pPFrame->Pop(CURRENT_THREAD);
 
+    return pbRetVal;
+}
+
+//=============================================================================
+// This function generates the real code for a method and installs it into
+// the methoddesc. Usually ***BUT NOT ALWAYS***, this function runs only once
+// per methoddesc. In addition to installing the new code, this function
+// returns a pointer to the new code for the prestub's convenience.
+//=============================================================================
+extern "C" PCODE STDCALL PreStubWorker(TransitionBlock* pTransitionBlock, MethodDesc* pMD)
```

Comment:
It would be nice to avoid introducing extra stack frame for the common PreStubWorker.